file-attribute-caching/src/main/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingFileSystemProvider.kt

```diff
@@ -224,7 +225,14 @@ internal class AttributeCachingFileSystemProvider : FileSystemProvider() {
         vararg options: LinkOption?
     ): V {
         val actualPath = path.getUnderlyingPath()
-        return actualPath.fileSystem.provider().getFileAttributeView(actualPath, type, *options)
+        return if (path is AttributeCachingPath && type is AclFileAttributeView) {
```

Comment:
Why are we handling this any differently than we would other views? `actualPath.fileSystem.provider().getFileAttributeView(actualPath, type, *options)` should work. And if we want the benefits of reading from cache, then the other attributes should probably also be doing the same thing.

Reply:
so for acl file attributes, I am caching the entire AclFileAttributeView rather than the attributes- this is very different than how the other attributes are currently cached (as they all extend BasicFileAttributes - which is not the view)

I have to cache the AclFileAttributeView because its difficult to translate the UserPrincipal owner and the AclEntry list by themselves to cachable variables (they'd have to be separate cached values). 
You cannot access the acl attributes via the provider function `fun <A : BasicFileAttributes?> readAttributes(path: Path,
        type: Class<A>,
        vararg options: LinkOption)` because thats for attributes that extend BasicFileAttributes only.
You can access acl attributes via `fun <V : FileAttributeView?> getFileAttributeView(}` or by `fun readAttributes(
        path: Path,
        attributes: String,
        vararg options: LinkOption
    )`

I need to do other_iops testing to see if caching the view truly works for performance, or if there is a performance hit everytime you query the view because the underlying filesystem is accessed to get the actual attributes.